### ISSUE TITLE
sacraficial like mythicals bugging out

### ISSUE THREAD

**Issue by Prosperity182** [SmartlyDressedGames/Unturned-3.x-Community#4220]
I do not know what causes this but for people outside it bugs out the sacrificial like mythicals.

https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/151689551/00dc7196-ac2f-45f0-9cb7-0735b995fce1

**Comment by MoltonMontro** [comment]
To clarify, the 3 mythical effects in this video are the **Crimson Navigator**, **Snowflake**, and **Sacrificial** mythics – correct?

**Comment by Prosperity182** [comment]
yes, it doesnt happen with crimson navigator I think its just a problem with snowflake and sacrificial

**Comment by AdamDN** [comment]
Did a bit of testing (mainly to get a better clip).
Seems like it's mostly Crimson Navigator that's causing it.

Snowflake + Sacrificial did have some very minor flickering:

https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/71527679/023f5266-7506-4687-b5f0-f81d71dc2c2e


But Snowflake + Crimson has major flickering:


https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/71527679/aa68504c-f064-4ed2-924b-4ae8ff68368e


And Crimson + Sacrificial has major flickering also:

https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/71527679/3e3a176d-9ba2-422f-94e3-f6da7a7f8772

**Comment by AdamDN** [comment]
Also, render mode Forward seems to fix it.

https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/71527679/4bf22bb4-f5e0-4dcd-98fa-2dd3d765dc64

**Comment by SDGNelson** [comment]
Thanks for the report, and thanks AdamDN for the additional videos! Sorry for my delay responding to this issue - I've received a couple support tickets about this as well so I had an item on the private issues list for it as well. I put it off because I had a bad feeling about it. 😨 

When trying to reproduce it, I found that it happens immediately with OpenGL but not with Direct3D or Vulkan. Are you using the OpenGL command-line option?

These effects are using the engine standard particle shader, so I tried updating to the latest LTS release. No luck there, unfortunately. Depending whether you're stuck with OpenGL we might be able to replace the shaders.

**Comment by AdamDN** [comment]
I don't think I'm seeing the same results. 
The videos above were all with no command lines. 

With command line `-force-opengl` I'm only seeing occasional artifacts around the top/edge of the screen (most of which weren't picked up by OBS) and they were happening on their own every 5-10 seconds ish (previously it required the camera to be moving):

https://github.com/SmartlyDressedGames/Unturned-3.x-Community/assets/71527679/83917a7c-bbe6-48ac-827f-6451b10bd070

`-force-vulkan` looks great, no issues with any combination of those three mythics.

**Comment by SDGNelson** [comment]
@AdamDN glad to hear it works well with Vulkan! Sorry, which platform are you on? (I searched your post history in the issue tracker but didn't find mention except that I think your servers at least are Windows.) My PC has an AMD GPU so I wonder if I'll get different results with an Nvidia GPU.

**Comment by AdamDN** [comment]
> [2023-12-13 22:50:38] Game version: 3.23.13.1 Engine version: 2021.3.29f1
> [2023-12-13 22:50:38] Requesting promo item grant
> [2023-12-13 22:50:38] Found Steam language 'english' in root Localization directory
> [2023-12-13 22:50:38] Platform: WindowsPlayer
> [2023-12-13 22:50:38] Operating System: Windows 11  (10.0.22621) 64bit
> [2023-12-13 22:50:38] System Memory: 32686MB
> [2023-12-13 22:50:38] Graphics Device Name: NVIDIA GeForce RTX 3090
> [2023-12-13 22:50:38] Graphics Device Type: Direct3D11
> [2023-12-13 22:50:38] Graphics Memory: 24340MB
> [2023-12-13 22:50:38] Graphics Multi-Threaded: False
> [2023-12-13 22:50:38] Render Threading Mode: LegacyJobified

Actually, my last message is wrong. I am seeing the same results as you, but they're Render Mode dependant.
I didn't realise I had swapped the render mode when I tested earlier. 

Direct3D + Deferred is how I clipped the original videos, but now I can't seem to recreate it 🤔. 
Direct3D + Forward is good.

OpenGL + Deferred is good. 
OpenGL + Forward has the occasional artifact. (clipped in last message)

Vulkan still looks good with both Deferred and Forward.
